examples/miral-shell/shell_main.cpp

```diff
@@ -83,37 +84,48 @@ int main(int argc, char const* argv[])
                 return true;
 
             case KEY_T:
-                external_client_launcher.launch({terminal_cmd});
+                {
+                    std::vector<std::string> command_line{terminal_cmd};
+                    if (terminal_cmd == "gnome-terminal" && boost::filesystem::exists("/usr/bin/gnome-terminal.real"))
+                    {
+                        // gnome-terminal is a horrid wrapper script on Ubuntu that needs frigging
+                        command_line.emplace_back("--disable-factory");
+                        command_line.emplace_back("--app-id");
+                        command_line.emplace_back("com.canonical.miral.Terminal");
+                    }
+
+                    external_client_launcher.launch(command_line);
+                }
                 return false;
 
-            case KEY_X:
-                external_client_launcher.launch_using_x11({"xterm"});
-                return false;
+                case KEY_X:
+                    external_client_launcher.launch_using_x11({"xterm"});
+                    return false;
 
-            default:
-                return false;
+                default:
+                    return false;
             };
         };
 
     Keymap config_keymap;
     DebugExtension debug_extensions;
 
     return runner.run_with(
-        {
-            CursorTheme{"default:DMZ-White"},
-            WaylandExtensions{},
-            X11Support{},
-            window_managers,
-            display_configuration_options,
-            external_client_launcher,
-            launcher,
-            config_keymap,
-            debug_extensions,
-            AppendEventFilter{quit_on_ctrl_alt_bksp},
-            StartupInternalClient{spinner},
-            pre_init(CommandLineOption{[&](std::string const& typeface) { ::wallpaper::font_file(typeface); },
-                                       "shell-wallpaper-font", "font file to use for wallpaper", ::wallpaper::font_file()}),
-            CommandLineOption{[&](std::string const& cmd) { terminal_cmd = cmd; },
-                              "shell-terminal-emulator", "terminal emulator to use", terminal_cmd}
-        });
+            {
+                    CursorTheme{"default:DMZ-White"},
+                    WaylandExtensions{},
+                    X11Support{},
+                    window_managers,
+                    display_configuration_options,
+                    external_client_launcher,
+                    launcher,
+                    config_keymap,
+                    debug_extensions,
```

Comment:
Excessive indentation

Reply:
Oops! I forgot to import the Mir layout options to CLion and it went wild!